### PR TITLE
nix: add script to update flake vendor hash

### PR DESCRIPTION
**Pull request by kradalby**
This PR adds a script to update the flake.nix vendor sha based on the current go.mod file. It also adds a note to the readme to do it after changing go.mod and go.sum.

### CODE REVIEW

File: update-flake.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# Updates SRI hashes for flake.nix.
```
Comment (maisem) [inline update-flake.sh:2]: I think this needs a copyright header

File: update-flake.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+# Copyright 2022 Tailscale Inc & Contributors
```
Comment (maisem) [inline update-flake.sh:2]: 2023

**Review by maisem — APPROVED** [review]

File: update-flake.sh (hunk 2 above)
Comment (willnorris) [reply update-flake.sh]: No date at all

Comment (maisem) [reply update-flake.sh]: yes, but we haven't updated this repo yet. I thought it would be better to be consistent and then clean up in a follow up.

Comment (maisem) [reply update-flake.sh]: (I don't feel strongly one way or another, whatever you recommend @willnorris)

**Pull request merged**